compiler/ballerina-compiler-api/src/main/java/org/ballerina/compiler/api/SemanticModel.java

```diff
@@ -50,6 +50,13 @@
      */
     Optional<Symbol> symbol(String srcFile, LinePosition position);
 
+    /**
+     * Retrieves the symbols of module-scoped constructs in the semantic model.
+     *
+     * @return A list of module-scoped symbols
+     */
+    List<Symbol> moduleLevelSymbols();
```

Comment:
Not entirely sure about this name. Please suggest any alternatives if you have any. This basically retrieves all the symbols in the module scope, both public and non-public.

Reply:
We need to  consider this along with, how are we going to define the term `Workspace` in ballerina. Until then, may be we can go with a name similar to `symbolsInModule`